compiler/src/iree/compiler/Codegen/Common/VectorizeMMT4d.cpp

```diff
@@ -1,173 +0,0 @@
-    }
-  }
-}
-
-/// Converts linalg.mmt4d into vector.contract.
-/// This converts linalg.mmt4d with operands <1x1xM0xK0>, <1x1xN0xK0>
-/// to vector.contract where K0 is the contraction dimension.
-struct VectorizeMMT4DOp : public OpRewritePattern<linalg::Mmt4DOp> {
-  using OpRewritePattern<linalg::Mmt4DOp>::OpRewritePattern;
-
-  LogicalResult matchAndRewrite(linalg::Mmt4DOp mmt4DOp,
-                                PatternRewriter &rewriter) const override {
-    Value lhs = mmt4DOp.inputs()[0];
-    Value rhs = mmt4DOp.inputs()[1];
-    Value dst = mmt4DOp.outputs()[0];
-
-    ShapedType lhsType = lhs.getType().dyn_cast<ShapedType>();
-    ShapedType rhsType = rhs.getType().dyn_cast<ShapedType>();
-    ShapedType dstType = dst.getType().dyn_cast<ShapedType>();
-
-    // This pattern expects tensors of static shapes.
-    // In practice, dynamic shapes are meant to be handled by other passes,
-    // ahead of this point. Dynamic outer dimensions (?x?xM0xK0) should be
-    // handled by a tiling pass typically running just ahead of the present
-    // pass. Dynamic inner dimensions (M1xK1x?x?) mean that the IR is not yet
-    // specialized to a specific SIMD ISA, and should be handled by dispatching
-    // to specialized code paths where these inner dimensions become static
-    // (M1xK1x?x? --> M1xK1xM0xK0)
-    if (!lhsType || !rhsType || !lhsType.hasStaticShape() ||
-        !rhsType.hasStaticShape()) {
-      return failure();
-    }
-
-    // We expect the incoming mmt4d to already have been maximally tiled, so
-    // that the outer dimensions are equal to 1.
-    {
-      int M1 = lhsType.getShape()[0];
-      int K1 = lhsType.getShape()[1];
-      int N1 = rhsType.getShape()[0];
-      if (M1 != 1 || K1 != 1 || N1 != 1) return failure();
-    }
-
-    // Read the inner dimensions.
-    int M0 = lhsType.getShape()[2];
-    int N0 = rhsType.getShape()[2];
-    int K0 = lhsType.getShape()[3];
-
-    Location loc = mmt4DOp.getLoc();
-    Value c0 = rewriter.create<arith::ConstantIndexOp>(loc, 0);
-
-    Type lhsElementType = lhsType.getElementType();
-    Type rhsElementType = rhsType.getElementType();
-    Type dstElementType = dstType.getElementType();
-
-    auto lhsVecType2D = VectorType::get({M0, K0}, lhsElementType);
-    auto rhsVecType2D = VectorType::get({N0, K0}, rhsElementType);
-    auto dstVecType2D = VectorType::get({M0, N0}, dstElementType);
-
-    auto lhs2DTensorType = RankedTensorType::get({M0, K0}, lhsElementType);
-    auto rhs2DTensorType = RankedTensorType::get({N0, K0}, rhsElementType);
-    auto dst2DTensorType = RankedTensorType::get({M0, N0}, dstElementType);
-
-    Value lhs2D = tensor::createCanonicalRankReducingExtractSliceOp(
-        rewriter, loc, lhs, lhs2DTensorType);
-    Value rhs2D = tensor::createCanonicalRankReducingExtractSliceOp(
-        rewriter, loc, rhs, rhs2DTensorType);
-    Value dst2D = tensor::createCanonicalRankReducingExtractSliceOp(
-        rewriter, loc, dst, dst2DTensorType);
```

Comment:
I suspect that these rank-reducing extract-slice ops that we used to do here, played a role in the patterns ultimately getting good codegen for the transfer_reads ?